larpandoradlcontent/LArHelpers/LArDLHelper.cc

```diff
@@ -0,0 +1,49 @@
+/**
+ *  @file   larpandoradlcontent/LArHelpers/LArDLHelper.cc
+ *
+ *  @brief  Implementation of the lar deep learning helper helper class.
+ *
+ *  $Log: $
+ */
+
+#include <larpandoradlcontent/LArHelpers/LArDLHelper.h>
+
+namespace lar_content
+{
+
+using namespace pandora;
+
+StatusCode LArDLHelper::LoadModel(const std::string &filename, LArDLHelper::TorchModel &model)
+{
+    try
+    {
+        model = torch::jit::load(filename);
+        std::cout << "Loaded the TorchScript model \'" << filename << "\'" << std::endl;
+    }
+    catch (...)
+    {
+        std::cout << "Error loading the TorchScript model \'" << filename << "\'" << std::endl;
+        return STATUS_CODE_FAILURE;
+    }
+
+    return STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArDLHelper::InitialiseInput(const at::IntArrayRef dimensions, TorchInput &tensor)
```

Comment:
So an `at::IntArrayRef` is, as perhaps the name suggests, something like a reference, so that we're not doing a big copy here?

Reply:
Yes, it's a constant reference to an array, although in this case, it's very unlikely to be a large array anyway, as this is ultimately defining the structure of the inputs to the network, which is usually of the form {batch_size, channels, y, x}.